analyzers/src/SonarAnalyzer.Common/Rules/ExpressionComplexityBase.cs

```diff
@@ -67,7 +67,7 @@ private int CalculateComplexity(SyntaxNode node)
 
             int CalculateComplexityRec(SyntaxNode node, int currentComplexity)
             {
-                if (node.Kind() is TSyntaxKind kind && ComplexityIncreasingKinds.Contains(kind))
+                if (Enum.ToObject(typeof(TSyntaxKind), node.RawKind) is TSyntaxKind kind && ComplexityIncreasingKinds.Contains(kind))
```

Comment:
[education] why is this necessary?

Reply:
this sounds like it should be an extension method?

Reply:
Added in [6ee063b](https://github.com/SonarSource/sonar-dotnet/pull/6500/commits/6ee063b01e76cc23772420b5f4bc27e1aa2f19fd). I'm not sure if we should keep it because it is on SyntaxNode and therefore quite noisy. An Extension method for RawKind is even worse because it is for `int`.

Reply:
sync: internal method, only for Base classes